src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -74,23 +74,37 @@ protected override void Shutdown()
             LaunchRequestArguments launchParams,
             RequestContext<object> requestContext)
         {
+            // Set the working directory for the PowerShell runspace to something reasonable
+            // such as the cwd passed in via launch.json. And in case that is null, use the
+            // folder or the script to be executed.
+            string workingDir = launchParams.Cwd ?? Path.GetDirectoryName(launchParams.Program);
+            var setWorkingDirCommand = new PSCommand();
+            setWorkingDirCommand.AddCommand(@"Microsoft.PowerShell.Management\Set-Location")
+                .AddParameter("LiteralPath", workingDir);
+
             // Execute the given PowerShell script and send the response.
             // Note that we aren't waiting for execution to complete here
             // because the debugger could stop while the script executes.
             Task executeTask =
                 editorSession.PowerShellContext
-                    .ExecuteScriptAtPath(launchParams.Program)
+                    .ExecuteCommand(setWorkingDirCommand)
```

Comment:
Instead of doing the ExecuteCommand here, I'd say do it in the block above with an await:

```csharp
            string workingDir = launchParams.Cwd ?? Path.GetDirectoryName(launchParams.Program);
            var setWorkingDirCommand = new PSCommand();
            setWorkingDirCommand.AddCommand(@"Microsoft.PowerShell.Management\Set-Location")
                 .AddParameter("LiteralPath", workingDir);

            await editorSession.PowerShellContext.ExecuteCommand(setWorkingDirCommand);
            
            Logger.Write(LogLevel.Verbose, "Working dir set to '" + workingDir + "'");
```

Using await adds some additional error handling for us that we don't get with ContinueWith. If the ExecuteCommand fails, the ContinueWith needs to check for the error and deal with it somehow.  Better to let the framework do that in this case I think.  Makes the code a bit simpler too.